waku/waku_archive/archive.nim

```diff
@@ -175,115 +164,33 @@ proc findMessages*(
   if rows.len == 0:
     return ok(ArchiveResponse(hashes: hashes, messages: messages, cursor: cursor))
 
-  ## Messages
   let pageSize = min(rows.len, int(maxPageSize))
 
-  #TODO once store v2 is removed, unzip instead of 2x map
-  #TODO once store v2 is removed, update driver to not return messages when not needed
-  if query.includeData:
-    topics = rows[0 ..< pageSize].mapIt(it[0])
-    messages = rows[0 ..< pageSize].mapIt(it[1])
+  hashes = rows[0 ..< pageSize].mapIt(it[0])
 
-  hashes = rows[0 ..< pageSize].mapIt(it[4])
+  if query.includeData:
+    topics = rows[0 ..< pageSize].mapIt(it[1])
+    messages = rows[0 ..< pageSize].mapIt(it[2])
 
-  ## Cursor
   if rows.len > int(maxPageSize):
     ## Build last message cursor
     ## The cursor is built from the last message INCLUDED in the response
     ## (i.e. the second last message in the rows list)
 
-    #TODO Once Store v2 is removed keep only message and hash
-    let (pubsubTopic, message, digest, storeTimestamp, hash) = rows[^2]
-
-    #TODO Once Store v2 is removed, the cursor becomes the hash of the last message
-    cursor = some(
-      ArchiveCursor(
-        digest: MessageDigest.fromBytes(digest),
-        storeTime: storeTimestamp,
-        sendertime: message.timestamp,
-        pubsubTopic: pubsubTopic,
-        hash: hash,
-      )
-    )
+    let (hash, _, _) = rows[^2]
+
+    cursor = some(hash)
 
-  # All messages MUST be returned in chronological order
+  # Messages MUST be returned in chronological order
   if not isAscendingOrder:
     reverse(hashes)
-    reverse(messages)
     reverse(topics)
+    reverse(messages)
 
   return ok(
-    ArchiveResponse(hashes: hashes, messages: messages, topics: topics, cursor: cursor)
+    ArchiveResponse(cursor: cursor, topics: topics, hashes: hashes, messages: messages)
   )
 
-proc findMessagesV2*(
```

Comment:
We are deprecating _StoreV2_ with that removal?

Reply:
Yes correct. Legacy archive still supports it.